common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationResultFactory.java

```diff
@@ -0,0 +1,133 @@
+                Logger.verbose(TAG, null, "User cancel the request in webview.");
+                result = createAuthorizationResultWithErrorResponse(AuthorizationStatus.USER_CANCEL,
+                        MicrosoftAuthorizationErrorResponse.USER_CANCEL, MicrosoftAuthorizationErrorResponse.USER_CANCELLED_FLOW);
+                break;
+
+            case AuthenticationConstants.UIResponse.BROWSER_CODE_COMPLETE:
+                final String url = data.getStringExtra(MSSTS_AUTHORIZATION_FINAL_URL);
+                result = parseUrlAndCreateAuthorizationResponse(url);
+                break;
+
+            case AuthenticationConstants.UIResponse.BROWSER_CODE_ERROR:
+                // This is purely client side error, possible return could be chrome_not_installed or the request intent is
+                // not resolvable
+                final String error = data.getStringExtra(ERROR_CODE);
+                final String errorDescription = data.getStringExtra(ERROR_DESCRIPTION);
+                result = createAuthorizationResultWithErrorResponse(AuthorizationStatus.FAIL, error, errorDescription);
+                break;
+
+            default:
+                result = createAuthorizationResultWithErrorResponse(AuthorizationStatus.FAIL,
+                        MicrosoftAuthorizationErrorResponse.UNKNOWN_ERROR, MicrosoftAuthorizationErrorResponse.UNKNOWN_RESULT_CODE);
+        }
+        return result;
+    }
+
+    private MicrosoftStsAuthorizationResult createAuthorizationResultWithErrorResponse(final AuthorizationStatus authStatus,
+                                                                                       final String error,
+                                                                                       final String errorDescription) {
+        Logger.info(TAG,"Error is returned from webview redirect");
+        Logger.infoPII(TAG,"error: " + error + " errorDescription: " + errorDescription);
+        AuthorizationErrorResponse errorResponse = new MicrosoftStsAuthorizationErrorResponse(error, errorDescription);
+        return new MicrosoftStsAuthorizationResult(authStatus, errorResponse);
+    }
+
+    private MicrosoftStsAuthorizationResult parseUrlAndCreateAuthorizationResponse(final String url) {
+        final HashMap<String, String> urlParameters = StringUtil.isEmpty(url) ? null : StringExtensions.getUrlParameters(url);
+        MicrosoftStsAuthorizationResult result;
+
+        if (urlParameters == null || urlParameters.isEmpty()) {
+            Logger.warn(TAG, null, "Invalid server response, empty query string from the webview redirect.");
+            result = createAuthorizationResultWithErrorResponse(AuthorizationStatus.FAIL,
+                    MicrosoftAuthorizationErrorResponse.AUTHORIZATION_FAILED,
+                    MicrosoftAuthorizationErrorResponse.AUTHORIZATION_SERVER_INVALID_RESPONSE);
+        }
+        else if (urlParameters.containsKey(CODE)) {
```

Comment:
Nit: This is super super super minor, but you wouldn't mind bringing the `else if` so it's formatted like:

```java
if (someCondition()) {
    // ...
} else if (someOtherCondition()) {
    // ...
}
```

Reply:
My bad, I misunderstood your comment of having a new line after "}" conditionals for readability and applied the changes here as well going against the standard IDE formatting. I was hesitant to do that but thought may be it's a pattern we follow in the team here. Glad to change it back like before.